badger/cmd/pick_table_bench.go

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2021 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime/pprof"
+	"sort"
+	"testing"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/dgraph-io/badger/v3/options"
+	"github.com/dgraph-io/badger/v3/table"
+	"github.com/dgraph-io/badger/v3/y"
+	"github.com/spf13/cobra"
+)
+
+var pickBenchCmd = &cobra.Command{
+	Use:   "picktable",
+	Short: "Benchmark pick tables.",
+	Long:  `This command simulates pickTables used in iterators.`,
+	RunE:  pickTableBench,
+}
+
+var (
+	pickOpts = struct {
+		readOnly   bool
+		sampleSize int
+		cpuprofile string
+	}{}
+	keys    [][]byte
+	handler levelHandler
+)
+
+func init() {
+	benchCmd.AddCommand(pickBenchCmd)
+	pickBenchCmd.Flags().BoolVar(
+		&pickOpts.readOnly, "read-only", true, "If true, DB will be opened in read only mode.")
+	pickBenchCmd.Flags().IntVar(
+		&pickOpts.sampleSize, "sample-size", 1000000, "Sample size of keys to be used for lookup.")
+	pickBenchCmd.Flags().StringVar(
+		&pickOpts.cpuprofile, "cpuprofile", "", "Write CPU profile to file.")
+}
+
+func pickTableBench(cmd *cobra.Command, args []string) error {
+	opt := badger.DefaultOptions(sstDir).
+		WithValueDir(vlogDir).
+		WithReadOnly(pickOpts.readOnly)
+	fmt.Printf("Opening badger with options = %+v\n", opt)
+	db, err := badger.OpenManaged(opt)
+	if err != nil {
+		return y.Wrapf(err, "unable to open DB")
+	}
+	defer db.Close()
```

Comment:
let's log the warning here in case close fails

Reply:
replaced with
```	
defer func() {
y.Check(db.Close())
}()
```